app/config/aggregation.go

```diff
@@ -4,20 +4,64 @@ import (
 	"crypto/tls"
 	"errors"
 	"fmt"
+	"slices"
 	"time"
 
 	"github.com/ClickHouse/clickhouse-go/v2"
 	"github.com/spf13/viper"
 )
 
+type AggregationEngine string
+
+const (
+	// Raw engine queries the raw events table
+	AggregationEngineClickHouseRaw AggregationEngine = "clickhouse_raw"
+	// MV engine maintains and queries materialized views
+	AggregationEngineClickHouseMV AggregationEngine = "clickhouse_mv"
+	// Parse engine parses events by meters and stores them in a separate unified meter events table
+	AggregationEngineClickHouseParse AggregationEngine = "clickhouse_parse"
+)
+
+func (e AggregationEngine) Values() []AggregationEngine {
+	return []AggregationEngine{AggregationEngineClickHouseRaw, AggregationEngineClickHouseMV, AggregationEngineClickHouseParse}
+}
+
+func (e AggregationEngine) Validate() error {
+	if !slices.Contains(e.Values(), e) {
+		return fmt.Errorf("invalid value")
+	}
+	return nil
+}
+
 type AggregationConfiguration struct {
+	// Engine is the aggregation engine to use
+	Engine     AggregationEngine
 	ClickHouse ClickHouseAggregationConfiguration
```

Comment:
Long term we should probably move the clickhouse config to top level and better organize engine configs.

This change only increases the chaos as it is with aggregation level, but clickhouse specific configs. :/